### ISSUE TITLE
Checking markdown READMEs exits with error, breaking pypi uploads

### ISSUE THREAD

**Issue by jstaf** [pypa/readme_renderer#107]
It looks like readme_renderer is improperly exiting with an error when reading markdown-formatted readme files (instead of exiting cleanly with exit code 0). This breaks all uploads to test.pypi.org and pypi.org that use markdown-formatted readmes.

```
$ python setup.py check -s -r
running check
warning: Check: Not checking long description content type 'text/markdown', this command only checks 'text/x-rst'.

error: Please correct your package.
$ echo $?
1
```

See https://github.com/pypa/warehouse/issues/4079 for more info.

**Comment by theacodes** [comment]
This behavior is *correct* here as you shouldn't use `setup.py check --restructuredtext` when you have a markdown readme. Will address in the warehouse issue.

**Comment by jstaf** [comment]
Hmmm, you are indeed right. Both myself and the other affected user in https://github.com/pypa/warehouse/issues/4079 appear to have gone down this documentation trail:

* `twine upload` fails with "HTTPError: 400 Client Error: The description failed to render in the default format of reStructuredText. See https://test.pypi.org/help/#description-content-type for more information. for url: https://test.pypi.org/legacy/"
* Visiting https://test.pypi.org/help/#description-content-type very heavily implies we should use `readme_renderer` to check `long_description` syntax.
* `readme_renderer` exits with "error: Please correct your package" when checking markdown files, leading users to believe that `readme_renderer`'s markdown parsing is what's blocking PyPi uploads.

With all of this in mind, it might be worthwhile for readme_renderer to exit cleanly without an error if the input is markdown (or update the documentation to make it explicitly clear that readme_renderer is not used for markdown files).

**Comment by Lapis-Hong** [comment]
so what is the solution????

**Comment by jstaf** [comment]
Upgrade your version of `wheel` + there was a bug that was fixed in pypa/warehouse#4079
